protofire/gnosis-conditional-exchange
# Fetch payouts from the blockchain

The payouts returned by the graph might be null for a closed market. To be sure they are correct, they are fetched from the blockchain now.

Besides that, there are some other changes here:
- The market returned by the graph might be null (for example, if there is no market for that id). The typing reflects that now.
- The payouts column shows `payout * shares` instead of `shares`, where `0 <= payout <= 1`.
- The closed market detail component was continously re-rendering because of a `useEffect` with an object dependency that changed in each render.

To test this:

1. Set the realitio timeout to a low value like 10. 9 or 11 should work too.
2. Create a question that opens immediately (same day, and 3 am or something like that)
3. Buy shares
4. Go to realitio and resolve the question
5. Go to the market page and check everything works
